uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/mocks/data/option-group-mock-data.js

```diff
@@ -0,0 +1,328 @@
+        "is_default": "0",
+        "weight": "7",
+        "is_optgroup": "0",
+        "is_reserved": "0",
+        "is_active": "1"
+      }, {
+        "id": "1112",
+        "option_group_id": "141",
+        "label": "1 3/4 Days",
+        "value": "1.75",
+        "name": "one_and_three_quarters_days",
+        "is_default": "0",
+        "weight": "8",
+        "is_optgroup": "0",
+        "is_reserved": "0",
+        "is_active": "1"
+      }],
+      hrleaveandabsences_leave_balance_change_type: [{
+        "id": "1099",
+        "option_group_id": "140",
+        "label": "Leave",
+        "value": "1",
+        "name": "leave",
+        "is_default": "0",
+        "weight": "0",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }, {
+        "id": "1100",
+        "option_group_id": "140",
+        "label": "Brought Forward",
+        "value": "2",
+        "name": "brought_forward",
+        "is_default": "0",
+        "weight": "1",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }, {
+        "id": "1101",
+        "option_group_id": "140",
+        "label": "Public Holiday",
+        "value": "3",
+        "name": "public_holiday",
+        "is_default": "0",
+        "weight": "2",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }, {
+        "id": "1102",
+        "option_group_id": "140",
+        "label": "Credit",
+        "value": "4",
+        "name": "credit",
+        "is_default": "0",
+        "weight": "3",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }, {
+        "id": "1103",
+        "option_group_id": "140",
+        "label": "Debit",
+        "value": "5",
+        "name": "debit",
+        "is_default": "0",
+        "weight": "4",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }, {
+        "id": "1104",
+        "option_group_id": "140",
+        "label": "Overridden",
+        "value": "6",
+        "name": "overridden",
+        "is_default": "0",
+        "weight": "5",
+        "is_optgroup": "0",
+        "is_reserved": "1",
+        "is_active": "1"
+      }]
+    },
+    /**
+     * Pick a random value out of a collection
+     *
+     * @param {array} the option group collection key
+     * @param {string} key - The sub-collection key
+     * @return {object}
+     */
+    randomValue: function (collection, key) {
+      var whichOptionGroup = this.mockedOptionValues[collection];
+      var randomOptionValue = _.sample(whichOptionGroup);
+      return randomOptionValue[key];
+    },
+    /**
+     * Gets all values for given key in collection object
+     *
+     * @param {array} the option group collection key
+     * @param {string} key - The sub-collection key
+     * @return {Array} of values
+     */
+    getAllValuesForKey: function (collection, key) {
```

Comment:
then we will be adding that private function (with same code) in each file whereever we will have to pick some random values